pyensembl/locus_with_genome.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2015-2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .locus import Locus
+
+class LocusWithGenome(Locus):
+    """
+    Common base class for Gene and Transcript to avoid copying
+    their shared logic.
+    """
+    def __init__(self, contig, start, end, strand, genome):
+        Locus.__init__(self, contig, start, end, strand)
+        self.genome = genome
+        self.db = self.genome.db
+
+    def to_dict(self):
+        return dict(
+            contig=self.contig,
+            start=self.start,
+            end=self.end,
+            strand=self.strand,
+            genome=self.genome._object_to_primitive_types())
+
+    @classmethod
+    def _initialize_nested_objects_in_state_dict(cls, state_dict):
+        state_dict["genome"] = cls._object_from_primitive_types(state_dict["genome"])
```

Comment:
Add a brief comment explaining what's happening here for when we look at it in a few weeks and forget everything?